packages/flutter_tools/lib/src/base/process.dart

```diff
@@ -614,19 +614,18 @@ class _DefaultProcessUtils implements ProcessUtils {
 }
 
 Future<int> exitWithHooks(int code, {required ShutdownHooks shutdownHooks}) async {
-  // Need to get the boolean returned from `messenger.shouldDisplayLicenseTerms()`
-  // before invoking the print welcome method because the print welcome method
-  // will set `messenger.shouldDisplayLicenseTerms()` to false
-  final FirstRunMessenger messenger =
-      FirstRunMessenger(persistentToolState: globals.persistentToolState!);
-  final bool legacyAnalyticsMessageShown =
-      messenger.shouldDisplayLicenseTerms();
-
-  // Prints the welcome message if needed for legacy analytics.
-  globals.flutterUsage.printWelcome();
-
```

Comment:
Apologies on the delayed review, slipped through the cracks

But couldn't we just check if we are on a bot before invoking the print statement for the welcome message? The `globals.analytics.shouldShowMessage` is not a good indicator for first run, it can return `true` if we had to change something within `package:unified_analytics` that would require us to show the consent message again

```dart
Future<int> exitWithHooks(int code, {required ShutdownHooks shutdownHooks}) async {
  // Need to get the boolean returned from `messenger.shouldDisplayLicenseTerms()`
  // before invoking the print welcome method because the print welcome method
  // will set `messenger.shouldDisplayLicenseTerms()` to false
  final FirstRunMessenger messenger =
      FirstRunMessenger(persistentToolState: globals.persistentToolState!);
  final bool legacyAnalyticsMessageShown =
      messenger.shouldDisplayLicenseTerms();

  // Prints the welcome message if needed for legacy analytics.
  // globals.flutterUsage.printWelcome();
  if (!(await globals.isRunningOnBot)) { // <--- Added
    globals.flutterUsage.printWelcome();
  }
```

Reply:
> You can test if this fix works by:

I was more wondering about the best way to **write** a test for it.

Reply:
> But couldn't we just check if we are on a bot before invoking the print statement for the welcome message?

That works for me - updated